platform/src/test/kotlin/org/stellar/anchor/platform/callback/RestRateIntegrationTest.kt

```diff
@@ -0,0 +1,273 @@
+
+    // Set up request and response
+    request =
+      gson.fromJson(
+        """
+        {
+          "type": "indicative",
+          "sell_asset": "iso4217:USD",
+          "sell_amount": "106",
+          "buy_asset": "stellar:USDC:GABCD"
+        }
+        """,
+        GetRateRequest::class.java,
+      )
+
+    rateResponse =
+      gson.fromJson(
+        """
+        {
+          "rate": {
+            "id": 1,
+            "price": "1.05",
+            "sell_amount": "100",
+            "buy_amount": "94.29",
+            "fee": {
+              "total": "1.00",
+              "asset": "iso4217:USD",
+              "details": [
+                {
+                  "name": "Sell fee #1",
+                  "description": "Fee related to selling the asset #1.",
+                  "amount": "0.70"
+                },
+                {
+                  "name": "Sell fee #2",
+                  "description": "Fee related to selling the asset #2.",
+                  "amount": "0.30"
+                }
+              ]
+            }
+          }
+        }
+        """,
+        GetRateResponse::class.java,
+      )
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = ["indicative", "firm"])
+  fun `test INDICATIVE and FIRM validateRateResponse`(type: String) {
+    request.type = from(type)
+    rateResponse.rate.id = "1234"
+    rateResponse.rate.expiresAt = Instant.now()
+
+    // The fee is in sell_asset
+    rateResponse.rate.fee.asset = usdAssetInfo.sep38AssetName
+    rateResponse.rate.sellAmount = "100.00"
+    rateResponse.rate.buyAmount = "94.29"
+    rateIntegration.validateRateResponse(request, rateResponse)
+
+    // The fee is in buy_asset
+    rateResponse.rate.fee.asset = usdcAssetInfo.sep38AssetName
+    rateResponse.rate.sellAmount = "100.00"
+    rateResponse.rate.buyAmount = "94.24"
+    rateIntegration.validateRateResponse(request, rateResponse)
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = ["-1.0", "-2000"])
+  @NullSource
+  fun `test bad sell and buy amounts`(badAmount: String?) {
+    // Bad sell amount
+    rateResponse.rate.sellAmount = badAmount
+    rateResponse.rate.buyAmount = "94.29"
+    var ex =
+      assertThrows<ServerErrorException> {
+        rateIntegration.validateRateResponse(request, rateResponse)
+      }
+    assertEquals(
+      "'sell_amount' is missing or not a positive number in the GET /rate response",
+      ex.message,
+    )
+
+    // Bad buy amount
+    rateResponse.rate.sellAmount = "100"
+    rateResponse.rate.buyAmount = badAmount
+    ex =
+      assertThrows<ServerErrorException> {
+        rateIntegration.validateRateResponse(request, rateResponse)
+      }
+    assertEquals(
+      "'buy_amount' is missing or not a positive number in the GET /rate response",
+      ex.message,
+    )
+  }
+
+  @Test
+  fun `test mis-matched sell_amount and buy_amount`() {
+    // Bad sell amount
+    rateResponse.rate.sellAmount = "100.01"
```

Comment:
It's hard to test why the test fails. Can we also set `buyAmount` and `fee` here and maybe add a comment with the expected amount?

Reply:
Added `buyAmount` and commented.